rfcs/0004-since-parameter.md

```diff
@@ -0,0 +1,193 @@
+
+### BFD API Details
+
+All the proposed API changes follow the FHIR standard. Keeping compability with the FHIR specification allows BFD clients to use existing FHIR libraries and tools.  
+
+The first improvement is a new `lastUpdated` field in the metadata object of a resource. This field contains the timestamp that the ETL process wrote resource to the BFD DB. Like all FHIR date fields, this timestamp must include the server's timezone \[[4](#ref4)\]. Resources based on records loaded before this RFC do not have a `lastUpdated` field. 
+
+The second change is a `_lastUpdated` query parameter for resource searches per the FHIR specification \[[5](#ref5)\]. FHIR specifies a large set of comparison operators. 
+BFD supports a subset of these operators: `eq`, `lt`, `le`, `gt` and `ge` operators. Two `_lastUpdated` parameters can be specified to form the upper and lower bounds of a time interval. 
+
+The BFD tracks the timestamp of the last write to the BFD database.
+The third change to BFD's API is returning the database last update timestamp in the `lastUpdated` field of Bundle resources. See the [Time Corner Cases](#time-corner-cases) section for more discussion on the motivation for this design decision. 
+
+If client asks for resources with a time interval that is beyond the database last update timestamp, the BFD will return all the resources it has that match and are bound inclusively by the database last update timestamp. From the BFD's perspective it is like the client asked for information from the future. 
+It is important for result consistency that the returned bundle does not return resources beyond the BFD's last updated timestamp.
+
+The BFD database records loaded before this RFC implementation will have a null `lastUpdated` field. The BFD treats these records as if they have a very early `lastUpdated` value. 
+Searches with a `_lastUpdated` parameter without a lower bound match these records; Likewise, searches with a lower bound never match these records. 
+This design allows a single query to retrieve both records with and without a `lastUpdate` metadata field. 
+
+### Single Beneficiary Implementors Details
+
+Partners can use the `_lastUpdated` parameter to get a single beneficiary's recently added FHIR resources. 
+To do this without missing a resource, the partner remembers (or has it's client remember) the returned Bundle's `lastUpdated` field. 
+This date represents the last update timestamp of the service database. All future updates to the database happen after this timestamp.
+
+![Last Updated](https://www.websequencediagrams.com/files/render?link=JJsgr8NxcInDVKyKqwtBiC5bTjdEafpXz4G1vbW2QbwtiveJ8IdHKl3r35lCdKdn)
+
+### Bulk Export Implementors Details
+
+BFD partner's which are implementing bulk export can add `_since` support. Just as it was in the single beneficiary case, keeping track of timestamps is important to avoid missing data. The following sequence diagram shows how this interaction should work. 
```

Comment:
Let's say a new EOB is added to the BFD DB. Will that update the `_lastUpdated` timestamp on the **Patient** resource to which the EOB belongs?

Reply:
The reason I am asking is to see if we could determine whether we can _completely_ avoid making a call to `/fhir/ExplanationOfBenefit?patient=...` by looking at `Patient.lastUpdated`, which would potentially make a significant positive impact on both ends.

Reply:
The literal answer to your question, the answer is no. A Patient's `lastUpdated` timestamp reflects the update time of the underlying beneficiary record from which the resource is derived. Adding new claim records doesn't change the beneficiary record or the resultant Patient resource. 

However, there is an optimization that you have in mind that I don't understand. Can you explain it more?

Reply:
Thanks for the answer, @RickHawesUSDS . Our interaction with BFD begins with obtaining a list of Patients by a given Part D contract number via an endpoint that @whytheplatypus has implemented. Then, for each Patient we search for their EOBs: a separate call per Patient. 

If we begin storing (read: caching) received EOBs on our end, then on subsequent runs _ideally_ we would like to be able to tell from a given Patient record whether or not their EOBs changed from the last time we fetched them, because that could save us from even making a network call altogether.

Does this make sense?

Reply:
Yes. This makes sense. So my understanding is that the AB2D to service keeps a cache of the EOBs received for a particular beneficiary. Periodically, the AB2D service would like to poll the BFD to receive any new EOBs for that particular beneficiary. Do I have this right? 

If so, this is an outline on how it can be accomplished: 

**Pre-step**: In addition to caching the EOB's for a particular beneficiary, you need to store one timestamp, the `transactionTime`, with each beneficiary.

**First-Time**: Fetch all EOBs for a particular beneficiary use the current EOB search end-point. The returned bundle of EOB has a `lastUpdated` timestamp. This is the `transactionTime` to store along-side the beneficiary's EOBs.  

**Periodically**: Fetch the EOB's that have arrived after the stored `transactionTime`. The new bundles will have an updated `transactionTime` to store as well. Repeat. 

```
v1/fhir/ExplanationOfBenefit?patient=<bene_id>&_lastUpdated=gt<transactionTime>
```